### PR TITLE
Component Button Fix

### PR DESCRIPTION
**Pull request by DhruvCopods**
Refactored button and checkbox input tags to Button and Checkbox Components

### CODE REVIEW

File: app/components/Logo.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const Logo = ({ height, width, styleClass }: LogoProps) => {
       src="../../assets/logo.svg"
       height={height}
       width={width}
-      alt="logo"
+      alt="k-quiz logo"
```
Comment (anuragP04) [inline app/components/Logo.tsx:13]: use constants

File: app/components/login/Login.tsx [hunk 2]
```diff
@@ -71,14 +71,15 @@ function Login({ actionData, redirectTo }: LoginProps) {
...
-          <div className="mt-6">
-            <input
+          <div className="mt-6 flex items-center justify-center">
+            <input type="hidden" name="redirectTo" value={redirectTo} />
+            <Button 
               tabIndex={0}
-              type="hidden"
-              name="redirectTo"
-              value={redirectTo}
+              buttonText="Sign in"
```
Comment (anuragP04) [inline app/components/login/Login.tsx:78]: use constants

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 3]
```diff
@@ -212,46 +213,43 @@ const AddQuestionInSection = () => {
...
 
-          <button
+          <Button 
             tabIndex={0}
-            id="saveAndAddMore"
-            disabled={transition.state === 'submitting'}
-            className={`flex h-9 items-center gap-1 rounded-lg bg-primary px-5 text-xs text-white ${
-              transition.state === 'submitting' && 'disabled:opacity-75'
-            }`}
+            id='saveAndAddMore'
```
Comment (anuragP04) [inline app/components/sections/add-question/AddQuestionInSection.tsx:242]: try using kebab case for ids

File: app/components/DeletePopUp.tsx [hunk 4]
```diff
@@ -76,29 +76,24 @@ export default function DeletePopUp({
...
+                      name='delete' 
+                      className='px-5'
+                      buttonText={commonConstants.delete} 
+                      onClick={handleDelete} />
                   </Form>
-                  <button
+                  <Button 
                     tabIndex={0}
+                    type='button' 
                     id="cancelDeletePopUp"
```
Comment (DhruvCopods) [inline app/components/DeletePopUp.tsx:92]: change this to kebab case

File: app/components/SideNavHeader.tsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { sideNav } from '~/constants/common.constants'
 const Header = ({ title }: { title: string }) => {
   return (
     <div className="flex items-center gap-4">
-      <img src={logo} alt="logo" />
+      <img src={logo} alt="k-quiz logo" />
```
Comment (DhruvCopods) [inline app/components/SideNavHeader.tsx:7]: use constant

File: app/components/members/AddMemberModal.tsx [hunk 6]
```diff
@@ -124,30 +125,24 @@ export default function AddMemberModal({
                     </div>
                   </div>
                   <div className="flex justify-end gap-2">
-                    <button
+                    <Button 
                       tabIndex={0}
-                      id="cancelAddButton"
-                      type="button"
-                      className="h-9 rounded-md px-4 text-sm text-gray-500"
+                      id='cancelAddButton'
```
Comment (DhruvCopods) [inline app/components/members/AddMemberModal.tsx:130]: change to kebab case

File: app/components/members/AddMemberModal.tsx [hunk 7]
```diff
@@ -124,30 +125,24 @@ export default function AddMemberModal({
...
-                    </button>
-                    <button
+                      varient='primary-outlined'
+                      buttonText={commonConstants.cancel} />
+                    <Button  
                       tabIndex={0}
-                      id="addbutton"
-                      type="submit"
-                      name="addMember"
+                      id='addbutton'
```
Comment (DhruvCopods) [inline app/components/members/AddMemberModal.tsx:138]: change to kebab case

File: app/components/members/MemberListItem.tsx [hunk 8]
```diff
@@ -55,24 +55,15 @@ export default function MemberListItem({
...
-            <Icon
-              onClick={openPopUp}
-              icon="ic:outline-delete-outline"
-              className={`pointer-cursor h-6 w-6 text-red-500 ${
-                loggedInUser && 'cursor-not-allowed text-red-300'
-              }`}
-            ></Icon>
-          </button>
+              <Icon
+                id='deleteButton'
```
Comment (DhruvCopods) [inline app/components/members/MemberListItem.tsx:59]: change to kebab case

File: app/components/members/MembersHeader.tsx [hunk 9]
```diff
@@ -20,14 +21,13 @@ export default function MembersHeader({
       <div className="col-span-full grid">
         <div className="flex items-center justify-between">
           <h1 className="text-3xl font-bold leading-9">{members.members}</h1>
-          <button
+          <Button
             tabIndex={0}
-            id="addMember"
-            className="h-9 rounded-lg bg-blue-900 px-4 text-xs font-medium leading-4 text-[#F0FDF4]"
+            id='addMember'
```
Comment (DhruvCopods) [inline app/components/members/MembersHeader.tsx:26]: change to kebab case

File: app/components/sections/AddSection.tsx [hunk 10]
```diff
@@ -85,31 +86,22 @@ const AddSection = ({
...
+                      tabIndex={0}
+                      type='button'
+                      className='h-9 px-4'
+                      onClick={() => setOpen(false)}
+                      varient="primary-outlined"
+                      buttonText={commonConstants.cancel} />
+                    <Button 
+                      tabIndex={0}
+                      type='submit'
+                      id='submitButton'
```
Comment (DhruvCopods) [inline app/components/sections/AddSection.tsx:99]: change to kebab case

File: app/components/sections/SectionDetails.tsx [hunk 11]
```diff
@@ -28,13 +29,13 @@ const SectionDetails = () => {
...
         >
-          <button
+          <Button
             tabIndex={0}
-            id="addQuestion"
-            className="h-9 rounded-lg bg-primary px-5 text-xs text-[#F0FDF4]"
-          >
-            + {addQuestion.addQuestion}
-          </button>
+            id='addQuestion'
```
Comment (DhruvCopods) [inline app/components/sections/SectionDetails.tsx:34]: change to kebab case

File: app/components/sections/add-question/AddQuestionInSection.tsx [hunk 12]
```diff
@@ -212,46 +213,43 @@ const AddQuestionInSection = () => {
...
         <div className="flex gap-2">
-          <button
+          <Button 
             tabIndex={0}
-            id="saveAndExit"
-            disabled={transition.state === 'submitting'}
-            className={`flex h-9 items-center gap-1 rounded-lg bg-primary px-5 text-xs text-white ${
-              transition.state === 'submitting' && 'disabled:opacity-75'
-            }`}
+            id='saveAndExit'
```
Comment (DhruvCopods) [inline app/components/sections/add-question/AddQuestionInSection.tsx:228]: change to kebab case

File: app/components/side-nav/CandidateSideNav.tsx [hunk 13]
```diff
@@ -36,7 +36,7 @@ const CandidateSideNav = ({
     <div className="flex h-full flex-col overflow-auto">
       <div className="p-5">
         <div className="flex items-center gap-4">
-          <img src={candidateLogo} alt="logo" height={'40px'} />
+          <img src={candidateLogo} alt="candidate logo" height={'40px'} />
```
Comment (DhruvCopods) [inline app/components/side-nav/CandidateSideNav.tsx:39]: use constant

File: app/components/SideNavHeader.tsx (hunk 5 above)
Comment (DhruvCopods) [inline app/components/SideNavHeader.tsx:7]: use constant

File: app/components/assessment/InvalidLink.tsx [hunk 14]
```diff
@@ -12,7 +12,7 @@ const InvalidLinkComponent = () => {
         </span>
       </div>
       <div className="mt-[-80px] flex h-full w-full flex-col items-center justify-center gap-4">
-        <img src={InvalidLink} alt="invalidLink" />
+        <img src={InvalidLink} alt="invalid link" />
```
Comment (DhruvCopods) [inline app/components/assessment/InvalidLink.tsx:15]: use constant

File: app/components/members/AddMemberModal.tsx (hunk 6 above)
Comment (DhruvCopods) [inline app/components/members/AddMemberModal.tsx:130]: change to kebab case

File: app/components/members/AddMemberModal.tsx (hunk 7 above)
Comment (DhruvCopods) [inline app/components/members/AddMemberModal.tsx:138]: change to kebab case

File: app/components/members/MemberListItem.tsx (hunk 8 above)
Comment (DhruvCopods) [inline app/components/members/MemberListItem.tsx:59]: change to kebab case

File: app/components/members/MembersHeader.tsx (hunk 9 above)
Comment (DhruvCopods) [inline app/components/members/MembersHeader.tsx:26]: change to kebab case

File: app/components/sections/AddSection.tsx (hunk 10 above)
Comment (DhruvCopods) [inline app/components/sections/AddSection.tsx:99]: change to kebab case

File: app/components/tests/AddTest.tsx [hunk 15]
```diff
@@ -144,85 +145,60 @@ const AddTestComponent = ({ sections }: { sections: Array<TestSection> }) => {
...
-            aria-label={commonConstants.backButton}
-          >
-            {commonConstants.backButton}
-          </button>
-          {currentTab != 2 ? (
-            <button
+            title='Previous Tab' 
+            className='h-9 px-7' 
+            varient='primary-solid'
+            id='backButton'
```
Comment (DhruvCopods) [inline app/components/tests/AddTest.tsx:173]: change to kebab case

File: app/components/tests/AddTest.tsx [hunk 16]
```diff
@@ -144,85 +145,60 @@ const AddTestComponent = ({ sections }: { sections: Array<TestSection> }) => {
...
-              onClick={() => setCurrentTab(currentTab + 1)}
-              disabled={!(name && description) || currentTab === 2}
-              aria-label='Next'
-            >
-              {commonConstants.nextButton}
-            </button>
+              title='Next Tab' 
+              className='h-9 px-7' 
+              varient='primary-solid'
+              id='nextButton'
```
Comment (DhruvCopods) [inline app/components/tests/AddTest.tsx:183]: change to kebab case

File: app/components/tests/TestList.tsx [hunk 17]
```diff
@@ -50,16 +53,15 @@ const TestList = ({
...
-          className="flex h-9 items-center rounded-lg bg-primary px-5 text-xs text-white"
-          title={testsConstants.addTestbutton}
-          aria-label={testsConstants.addTestbutton}
-        >
-          + {testsConstants.addTestbutton}
-        </Link>
+        <Button 
+        className='px-5'
+        onClick={() => navigate('/tests/add-test')}
+        id='addTest'
```
Comment (DhruvCopods) [inline app/components/tests/TestList.tsx:59]: change to kebab case

File: cypress/e2e/add-questions/AddQuestion.cy.ts [hunk 18]
```diff
@@ -448,7 +448,7 @@ describe('Test for section-details', () => {
         .then((len) => {
           lengthBefore = len
         })
-      cy.get('button.flex').should('have.text', '+ Add Options').click()
+      cy.get('button.px-5').should('have.text', '+ Add Options').click()
```
Comment (DhruvCopods) [inline cypress/e2e/add-questions/AddQuestion.cy.ts:451]: use id rather than class

**Pull request closed without merging**